yodeploy/ipc_logging.py

```diff
@@ -49,14 +53,25 @@ def handle(self):
                 break
 
 
-class ThreadedLogStreamServer(SocketServer.ThreadingMixIn,
-                              SocketServer.UnixStreamServer):
+class ThreadedLogStreamServer(socketserver.ThreadingMixIn,
+                              socketserver.UnixStreamServer):
     '''A Server that will receive all log messages sent over the supplied
     (already open) socket
     '''
     def __init__(self):
-        self.socket, self.remote_socket = socket.socketpair(socket.AF_UNIX,
-                                                            socket.SOCK_STREAM)
+        self.socket, self.remote_socket = socket.socketpair(
+            socket.AF_UNIX, socket.SOCK_STREAM)
+
+        # In python >= 3.4, file descriptors are not inheritable by child
+        # processes by default. We need to be able to pass the fd for
+        # self.remote_socket to child processes via a command line argument
+        # in various scenarios, so here we explicitly configure the fd to allow
+        # for that.
+        try:
+            self.remote_socket.set_inheritable(True)
+        except AttributeError:  # set_inheritable not available in python < 3.4
```

Comment:
Not available because it's not necessary. I'd be tempted to say: `only disabled under python >= 3.4`